### PR TITLE
Improve documentation of the preview mode

### PR DESCRIPTION
**Pull request by hoel-bagard**
## Summary

This PR was prompted by the discussion in #10153.
It adds CLI tab examples next to the `pyproject.toml` and `ruff.toml`  examples. It should be helpful for users wanting to try out the preview mode without modifying or creating a  `.toml` file.
It also adds a paragraph to try to make the effect of the preview mode less confusing.

### CODE REVIEW

**Comment by hoel-bagard** [comment]
@buhtz does this added paragraph makes it less misleading in your opinion ?

**Comment by buhtz** [comment]
It is an improvement.

Please see https://docs.astral.sh/ruff/rules/too-many-blank-lines/

> This rule is unstable and in [preview](https://docs.astral.sh/ruff/preview/). The --preview flag is required for use.

This is also misleading. I would suggest something like this (in all preview rules):

> The two switches `--select` and `--preview` are required for use.

**Comment by hoel-bagard** [comment]
The `--select` part is required for nearly all rules (by default only `["E4", "E7", "E9", "F"]` are selected), so I don't think it makes sense to specify it only on preview rules.

**Comment by buhtz** [comment]
> The `--select` part is required for nearly all rules (by default only `["E4", "E7", "E9", "F"]` are selected), so I don't think it makes sense to specify it only on preview rules.

But the docs state different. The docs say that you can use the `--preview` switch and all unstable rules are active. That is how some readers might understand it. And it would make sense by the way. My intention was to activate all unstable-rules to support your development via testing that rules.

File: docs/preview.md [hunk 1]
```diff
@@ -3,8 +3,9 @@
 Ruff includes an opt-in preview mode to provide an opportunity for community feedback and increase confidence that
 changes are a net-benefit before enabling them for everyone.
 
-Preview mode enables a collection of newer lint rules, fixes, and formatter style changes that are
-considered experimental or unstable,.
+Preview mode enables a collection of new lint rules, fixes, formatter style and other changes that are considered experimental or unstable.
+
+Simply enabling preview mode does not imply that all the new preview features/changes will be turned on. Some, like the rules that are in preview, need to be turned on explicitly (see the [rules section](#using-rules-that-are-in-preview). Others, like the formatter style or default values (for example `output-format`) will be turned on by simply enabling the preview mode.
```
Comment (MichaReiser) [inline docs/preview.md:8]: Nit: I would avoid explicitly naming `output-format` here. We otherwise have to come up with a new preview feature everytime we remove the feature mentioned in the documentation.

**Review by MichaReiser — APPROVED** [review]
Nice! I like it. I let @zanieb have a look at it too.

**Comment by MichaReiser** [comment]
> The two switches --select and --preview are required for use.

Whether you need to enable the rule using `select`, `extend-select` or similar depends on your current configuration. Using `--select` (or similar) is also not specific to `--preview` but applies to all rules. That's why what we have now makes more sense to me.

File: docs/preview.md (hunk 1 above)
Comment (hoel-bagard) [reply docs/preview.md]: I hesitated when writing it, better keep it general indeed.
Removed in b0a9db688.

File: docs/preview.md [hunk 2]
```diff
@@ -63,6 +77,13 @@ config file:
     extend-select = ["HYP001"]
     ```
 
+=== "CLI"
+
+    ```console
+    ruff check . --select HYP001
```
Comment (zanieb) [inline docs/preview.md:83]: The `.` is superfluous in all of these examples, that's the default behavior these days.

Comment (zanieb) [reply docs/preview.md:83]: We're using `extend-select` in all of the other examples, why use `--select` here?

File: docs/preview.md [hunk 3]
```diff
@@ -43,11 +51,17 @@ To enable preview style formatting without enabling any preview lint rules:
...
+    ```console
+    ruff format . --preview
+    ```
+
 ## Using rules that are in preview
 
 If a rule is marked as preview, it can only be selected if preview mode is enabled. For example, consider a
 hypothetical rule, `HYP001`. If `HYP001` were in preview, it would _not_ be enabled by adding following to your
-config file:
+config file or only using the `--select` flag on the CLI:
```
Comment (zanieb) [inline docs/preview.md:64]: If you want to add more here, I'd rephrase entirely

> If `HYP001` were in preview, it would _not_ be enabled by adding it to the selected rules.

File: docs/preview.md [hunk 4]
```diff
@@ -95,7 +123,13 @@ Similarly, it would _not_ be enabled via the `ALL` selector:
...
     ```
 
-However, it would be enabled in any of the above cases if you enabled preview in your configuration file:
+=== "CLI"
+
+    ```console
+    ruff check . --select ALL
+    ```
+
+However, it would be enabled in any of the above cases if you enabled preview in your configuration file or provided the `--preview` CLI flag:
```
Comment (zanieb) [inline docs/preview.md:132]: Similarly, I'd just generalize these e.g.

```suggestion
However, it would be enabled in any of the above cases if you enabled preview mode:
```

File: docs/preview.md [hunk 5]
```diff
@@ -3,8 +3,9 @@
 Ruff includes an opt-in preview mode to provide an opportunity for community feedback and increase confidence that
 changes are a net-benefit before enabling them for everyone.
 
-Preview mode enables a collection of newer lint rules, fixes, and formatter style changes that are
-considered experimental or unstable,.
+Preview mode enables a collection of new lint rules, fixes, formatter style and other changes that are considered experimental or unstable.
+
+Simply enabling preview mode does not imply that all the new preview features/changes will be turned on. Some, like the rules that are in preview, need to be turned on explicitly (see the [rules section](#using-rules-that-are-in-preview). Others, like the formatter style or default values changes will be turned on by simply enabling the preview mode.
```
Comment (zanieb) [inline docs/preview.md:8]: I find this a little confusing. We have a flag that determines if preview rules require "explicit" selection but by default they do not i.e. "implicit" selection of the rule via a prefix is sufficient. I'd be much more direct here:

```suggestion
Enabling preview mode does not select all preview rules by default. See the [rules section](#using-rules-that-are-in-preview) for details on selecting preview rules.
```

I'd also consider putting it in a note admonition.

File: docs/preview.md [hunk 6]
```diff
@@ -3,8 +3,9 @@
 Ruff includes an opt-in preview mode to provide an opportunity for community feedback and increase confidence that
 changes are a net-benefit before enabling them for everyone.
 
-Preview mode enables a collection of newer lint rules, fixes, and formatter style changes that are
-considered experimental or unstable,.
+Preview mode enables a collection of new lint rules, fixes, formatter style and other changes that are considered experimental or unstable.
```
Comment (zanieb) [inline docs/preview.md:6]:
```suggestion
Preview mode enables a collection of unstable features such as new lint rules and fixes, formatter style changes, interface updates, and more. Warnings about deprecated features may turn into errors when using preview mode.
```

File: docs/preview.md (hunk 2 above)
Comment (hoel-bagard) [reply docs/preview.md]: > The . is superfluous in all of these examples, that's the default behavior these days.

I chose to use the `.` to be consistent with the rest of the documentation (for example, see [the tutorial page](https://docs.astral.sh/ruff/tutorial/)). If it's better to remove the `.`, then I can open another PR to remove it from the rest of the documentation, what do you think ?

> We're using extend-select in all of the other examples, why use --select here?

It was a mistake, I fixed it in 9f5d9bcdf.

File: docs/preview.md (hunk 3 above)
Comment (hoel-bagard) [reply docs/preview.md]: Done in 9b6b7bd19.

**Review by buhtz — COMMENTED** [review]
I like that solution.

File: docs/preview.md (hunk 2 above)
Comment (zanieb) [reply docs/preview.md]: Sure we can address that separately, thanks!

Comment (hoel-bagard) [reply docs/preview.md]: Ok, I'll do that then. I've removed the `.` in this commit [356c4b0](https://github.com/astral-sh/ruff/pull/10168/commits/356c4b0580f63a918936a04b061b6503a28470d0).
